Docs/source/install/hpc/lxplus.rst

```diff
@@ -0,0 +1,141 @@
+.. _building-lxplus:
+
+LXPLUS (CERN)
+=============
+
+The LXPLUS cluster is located at CERN.
+
+* `Lxplus documentation <https://lxplusdoc.web.cern.ch>`__
+* Batch system: `HTCondor <https://batchdocs.web.cern.ch/index.html>`__
+* Filesystem locations:
+    * User folder: `/afs/cern.ch/user/<a>/<account>` (10GByte)
+    * Work folder: `/afs/cern.ch/work/<a>/<account>` (100GByte)
+    * Eos storage: `/eos/home-<a>/<account>` (1T)
+
+Through LXPLUS we have access to CPU and GPU nodes (the latter equipped with NVIDIA A100 or T4 GPUs).
+
+Installation
+------------
+Only very little software is pre-installed on Lxplus so we show how to install from scratch all the dependencies using Spack.
+
+For size reasons it is not advisable to install WarpX in the $HOME directory, while it should be installed in the "work directory". For this purpose we set an environment variable with the path to the "work directory"
+
+.. code-block:: bash
+
+    export WORK=/afs/cern.ch/work/<a>/<account>/
+
+GCC
+^^^
+The pre-installed GNU compiler is outdated so we need a more recent compiler. Here we use the gcc 9.2.0 from the LCG project, but other options are possible.
+
+We activate it by doing
+
+.. code-block:: bash
+
+    source /cvmfs/sft.cern.ch/lcg/releases/gcc/9.2.0/x86_64-centos7/setup.sh
+
+In order to avoid using different compilers this line could be added directly into the ``$HOME/.bashrc`` file.
+
+Spack
+^^^^^
+We download and activate Spack in ``$WORK``:
+
+.. code-block:: bash
+
+    cd $WORK
+    git clone -c feature.manyFiles=true https://github.com/spack/spack.git
+    source spack/share/spack/setup-env.sh
+
+When installing packages Spack will try to set permissions in a way which is forbidden on the LXPLUS file system (AFS), but we can avoid this by modifying the ``spack/etc/spack/defaults/config.yaml``. Inside this file we need to change the line
+
+.. code-block:: bash
+
+    allow_sgid: true
+
+to
+
+.. code-block:: bash
+
+    allow_sgid: false
+
+Now we add our gcc 9.2.0 compiler to spack:
+
+.. code-block:: bash
+
+    spack compiler find /cvmfs/sft.cern.ch/lcg/releases/gcc/9.2.0-afc57/x86_64-centos7/bin/
+
+Installing the Dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+With the following commands we install the dependencies for WarpX:
+
+.. code-block:: bash
+
+    spack install ncurses+symlinks %gcc@9.2.0
```

Comment:
Individual `spack install` + `spack load` commands can potentially load incompatible software. A better way to install this is using a virtual environment, [like here](https://warpx.readthedocs.io/en/latest/install/dependencies.html#spack-macos-linux).

All packages and the option + compiler above can be summarized in a single "download this `.yaml` environment file", which might be easiest to do here.

Example of such a file: https://github.com/openPMD/openPMD-api/blob/dev/.github/ci/spack-envs/clang8_py38_mpich_h5_ad1_ad2/spack.yaml

- `specs:` section - the packages I want
- `packages:` section - mainly selecting what I want to re-use from existing packages on the system
- `compilers:` section - the system compiler I like to configure & use
- `config:` section - currently only used to limit the -j compilation to two cores

In order to populate the `compilers:` section, you can copy out from `spack config edit compilers` in your current setup.

Using this file will also allow us to make `gcc@9.2.0` the default compiler, so we don't need to type it all the time, by making it the first and only entry in `compilers:`.

For CUDA, you could either do an additional environment file or add an option like is done in #2719 to add the additional CUDA package and variants.